python/lsst/validate/drp/io.py

```diff
@@ -0,0 +1,57 @@
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+from __future__ import print_function, division, absolute_import
+
+import json
+import numpy as np
+
+
+def saveKpmToJson(KpmStruct, filename):
```

Comment:
@ktl  For context, I understand the motivations behind your questions and don't really disagree.

But for the purposes of this Pull Request:
As part of DM-2050 I was told by @frossie to have `validate_drp` output to JSON, so that's what I did.  Happy to have a further discussion about how DM-4957 should have been implemented, but it's out of scope for the Pull Request.

A few notes:
1. `json` is in the Python standard library.  There is no additional dependency being added by using JSON.
2. JSON is a subset of YAML.
3. I got a bit confused myself about YAML vs. JSON when answering @timj 's question above about implementing this in pipe.base.Struct (which is also beyond scope).
4. There is no current "other side of this interface" (DM-2050).